spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -500,33 +500,32 @@ protected PartitioningWriter<PositionDelete<InternalRow>, DeleteWriteResult> new
             targetFileSize,
             deleteGranularity,
             rewritableDeletes != null
-                ? new PreviousDeleteLoader(table, rewritableDeletes)
+                ? new PreviousDeleteLoader(table, rewritableDeletes, specs)
                 : path -> null /* no previous file scoped deletes */);
       }
     }
   }
 
   private static class PreviousDeleteLoader implements Function<CharSequence, PositionDeleteIndex> {
-    private final Map<String, DeleteFileSet> deleteFiles;
+    private final RewritableDeletes rewritableDeletes;
     private final DeleteLoader deleteLoader;
+    private final Map<Integer, PartitionSpec> specs;
 
-    PreviousDeleteLoader(Table table, Map<String, DeleteFileSet> deleteFiles) {
-      this.deleteFiles = deleteFiles;
+    PreviousDeleteLoader(
+        Table table, RewritableDeletes rewritableDeletes, Map<Integer, PartitionSpec> specs) {
+      this.rewritableDeletes = rewritableDeletes;
       this.deleteLoader =
           new BaseDeleteLoader(
               deleteFile ->
                   EncryptingFileIO.combine(table.io(), table.encryption())
                       .newInputFile(deleteFile));
+      this.specs = specs;
     }
 
     @Override
     public PositionDeleteIndex apply(CharSequence path) {
-      DeleteFileSet deleteFileSet = deleteFiles.get(path.toString());
-      if (deleteFileSet == null) {
-        return null;
-      }
-
-      return deleteLoader.loadPositionDeletes(deleteFileSet, path);
+      return deleteLoader.loadPositionDeletes(
+          rewritableDeletes.deletesFor(path.toString(), specs), path);
```

Comment:
I'll go back to returning a null index in case there are no deletes for the given path, the internal implementation of loader takes care of it regardless but it isn't obvious without reading into it.